src/richie/apps/core/cache.py

```diff
@@ -131,3 +135,41 @@ def close(self, **kwargs):
         del self._client_read
         self._client_write = None
         self._client_read = None
+
+
+class LimitBrowserCacheTTLHeaders(MiddlewareMixin):
+    """
+    This middleware allows to define a maximum cache timeout for the cache
+    response headers (Cache-control: max-age and Expires).
+
+    It can be configured with the MAX_BROWSER_CACHE_TTL setting.
+    This value is expressed in seconds.
+
+    LimitBrowserCacheTTLHeaders should be placed at the beginning of the
+    MIDDLEWARE list, so that it'll get called last during the response phase.
+    """
+
+    # pylint: disable=no-self-use
+    def process_response(self, request, response):
+        """
+        Rewrite the "Cache-control" and "Expires headers" in the response
+        if needed.
+        """
+        max_ttl = getattr(settings, "MAX_BROWSER_CACHE_TTL", None)
+        try:
+            max_ttl = int(max_ttl)
+            if max_ttl < 0:
+                raise ValueError("MAX_BROWSER_CACHE_TTL must be a positive integer")
+        except (ValueError, AttributeError, TypeError) as err:
```

Comment:
can you really get all these errors int casting?

Reply:
The `TypeError` and `ValueError` are raised by `int()`
And the `AttributeError` was for the  `getattr` that I forgot to move in the `try:` block :man_facepalming:

Reply:
Ok, here is the fixed version, with another test for the case where `MAX_BROWSER_CACHE_TTL` is not defined.